lib/cloud-metadata/azure.js

```diff
@@ -92,22 +88,21 @@ function formatMetadataStringIntoObject (string) {
     machine: {
       type: null
     },
-    provider: null,
+    provider: 'azure',
     region: null
   }
   const parsed = JSON.parse(string)
   if (!parsed.compute) {
     return metadata
   }
   const data = parsed.compute
-  metadata.account.id = data.subscriptionId + ''
-  metadata.instance.id = data.vmId + ''
-  metadata.instance.name = data.name + ''
-  metadata.project.name = data.resourceGroupName + ''
-  metadata.availability_zone = data.zone + ''
-  metadata.machine.type = data.vmSize + ''
-  metadata.region = data.location + ''
-  metadata.provider = 'azure'
+  metadata.account.id = String(data.subscriptionId)
+  metadata.instance.id = String(data.vmId)
+  metadata.instance.name = String(data.name)
+  metadata.project.name = String(data.resourceGroupName)
+  metadata.availability_zone = String(data.zone)
+  metadata.machine.type = String(data.vmSize)
+  metadata.region = String(data.location)
```

Comment:
nit: Could these assignments move into the `const metadata = ` declaration?

Reply:
Good callout -- but for this specific case if the returned metadata doesn't contain a `.compute`  we still want to return some sort of object that identifies the provider.  Will leave this nit undone.

Reply:
Ah, I see. You are right. Thanks.